tests/Sulu/Component/Rest/ListBuilder/Metadata/FieldDescriptorFactoryTest.php

```diff
@@ -0,0 +1,183 @@
+            'lastName' => ['name' => 'lastName', 'translation' => 'contact.contacts.lastName', 'default' => true],
+            'avatar' => [
+                'name' => 'avatar',
+                'translation' => 'public.avatar',
+                'default' => true,
+                'type' => 'thumbnails',
+                'sortable' => false
+            ],
+            'fullName' => [
+                'instance' => DoctrineConcatenationFieldDescriptor::class,
+                'name' => 'fullName',
+                'translation' => 'public.name',
+                'disabled' => true,
+                'sortable' => false,
+                'class' => 'test-class',
+                'minWidth' => '50px',
+                'width' => '100px',
+                'select' => 'CONCAT(SuluContactBundle:Contact.firstName, CONCAT(\' \', SuluContactBundle:Contact.lastName))'
+            ],
+            'city' => ['name' => 'city', 'translation' => 'contact.address.city', 'default' => true],
+        ];
+
+        $this->assertFieldDescriptors($expected, $fieldDescriptor);
+    }
+
+    public function testGetFieldDescriptorForClassMinimal()
+    {
+        $this->locator->findFileForClass(new \ReflectionClass(new \stdClass()), 'xml')
+            ->willReturn(__DIR__ . '/Resources/minimal.xml');
+
+        $provider = new ChainProvider($this->chain);
+        $factory = new FieldDescriptorFactory($provider, $this->configCache->reveal());
+        $fieldDescriptor = $factory->getFieldDescriptorForClass(\stdClass::class);
+
+        $this->assertEquals(['id', 'firstName', 'lastName'], array_keys($fieldDescriptor));
+
+        $expected = [
+            'id' => ['name' => 'id', 'translation' => 'public.id', 'disabled' => true, 'type' => 'integer'],
+            'firstName' => ['name' => 'firstName', 'translation' => 'contact.contacts.firstName', 'default' => true],
+            'lastName' => ['name' => 'lastName', 'translation' => 'contact.contacts.lastName', 'default' => true],
+        ];
+
+        $this->assertFieldDescriptors($expected, $fieldDescriptor);
+    }
+
+    public function testGetFieldDescriptorForClassEmpty()
+    {
+        $this->locator->findFileForClass(new \ReflectionClass(new \stdClass()), 'xml')
+            ->willReturn(__DIR__ . '/Resources/empty.xml');
+
+        $provider = new ChainProvider($this->chain);
+        $factory = new FieldDescriptorFactory($provider, $this->configCache->reveal());
+        $fieldDescriptor = $factory->getFieldDescriptorForClass(\stdClass::class);
+
+        $this->assertEmpty($fieldDescriptor);
+    }
+
+    protected function assertFieldDescriptors(array $expected, array $fieldDescriptors)
+    {
+        foreach ($expected as $name => $expectedData) {
+            $this->assertFieldDescriptor($expectedData, $fieldDescriptors[$name]);
+        }
+    }
+
+    protected function assertFieldDescriptor(array $expected, FieldDescriptorInterface $fieldDescriptor)
```

Comment:
I would default to making methods `private` - it is probably not a good thing to extend this class - if another class were going to use these methods it would be better to move them to an abstract class. i.e. it is better not to encourage bad practice by implying that the class is extendable :) But not very important.